app/src/lib/editors/darwin.ts

```diff
@@ -131,15 +131,28 @@ async function findApplication(
 ): Promise<string | null> {
   for (const identifier of editor.bundleIdentifiers) {
     try {
-      const installPath = await appPath(identifier)
-      const exists = await pathExists(installPath)
-      if (exists) {
+      // app-path not finding the app isn't an error, it just means the
+      // bundle isn't registered on the machine.
+      // https://github.com/sindresorhus/app-path/blob/0e776d4e132676976b4a64e09b5e5a4c6e99fcba/index.js#L7-L13
+      const installPath = await appPath(identifier).catch(e =>
+        e.message === "Couldn't find the app"
```

Comment:
Looking at the source code, wouldn't be better to check `e.code === 2`? If I'm not wrong, that 2 comes from `ENOENT` so you can probably even use the `os.constants.errno.ENOENT` constant from Node.

I want to think that would be a bit less magic and a bit less coupled to the implementation of app-path than checking the "user-facing" string 😳  What do you think?

Reply:
Although ENOENT is likely what inspired it, it seems the return value from the app-path executable doesn't use the constant and is just as magic. See https://github.com/sindresorhus/app-path/blob/0e776d4e132676976b4a64e09b5e5a4c6e99fcba/main.m#L21-L23

So I think either option is tightly coupled with the implementation of app-path although matching the string is perhaps more readable? I'm like 2 out of 10 on the care-scale here though so I'm happy to switch to checking `code` instead if you'd like but I'm not sure using the constant variable is merited, what do you think?

Reply:
Ohhh that's sad… I'm ok with leaving it as it is!